firebase.go

```diff
@@ -99,6 +99,11 @@ func New(url string) *Firebase {
 		CheckRedirect: redirectPreserveHeaders,
 	}
 
+	return NewFromHttpClient(url, client)
+}
+
+// NewFromHttpClient creates a new Firebase reference with given http client
+func NewFromHttpClient(url string, client *http.Client) *Firebase {
```

Comment:
I agree with you on not exposing the client @campoy . I'd rather not worry about the Client potentially being `nil`.

I'd just vote for replacing the old constructor and just calling it `New`. If the user passes a `nil` client, we can set it to the `DefaultClient`